main.js

```diff
@@ -53,64 +55,42 @@ function generateRandomPalette() {
 	}
 }
 
-function updatePalette(array) {
-	for (var i = 0; i < array.length; i++) {
-		allBoxes[i].style.backgroundColor = array[i];
-		allHexes[i].innerText = array[i];
+function updatePalette(colorValues) {
+	for (var i = 0; i < colorValues.length; i++) {
+		allBoxes[i].style.backgroundColor = colorValues[i];
+		allHexes[i].innerText = colorValues[i];
 	}
 }
 
 function savePalette() {
-	var paletteCopy = currentPalette.slice();
-	savedPalettes.push(paletteCopy);
+	var savedPalette = currentPalette.slice();
+	savedPalettes.push(savedPalette);
 	return savedPalettes;
 }
 
 
 function displaySavedPalettes() {
 	miniPaletteContainer.innerHTML = '';
 	for (var i = 0; i < savedPalettes.length; i++) {
-	miniPaletteContainer.innerHTML += `
-	<section class="mini-palette">
-		<div class="mini-box" style="background-color:${savedPalettes[i][0]};">
-		</div>
-		<div class="mini-box" style="background-color:${savedPalettes[i][1]};">
-		</div>
-		<div class="mini-box" style="background-color:${savedPalettes[i][2]};">
-		</div>
-		<div class="mini-box" style="background-color:${savedPalettes[i][3]};">
-		</div>
-		<div class="mini-box" style="background-color:${savedPalettes[i][4]};">
-		</div>
-		<button class="delete-button" data-index="${i}">
-			<img src="assets/delete.png" alt="Delete Button">
-		</button>
-	</section>
-	`
+			var miniPalette = document.createElement('section');
```

Comment:
Made this function more dynamic and with less inline HTML. Created a nested for loop to create HTML elements for mini palette sections and mini boxes.